lib/sidebar-manager.js

```diff
@@ -329,6 +346,23 @@ exports.sidebarManager = {
                     }
                 });
 
+                // if we're notified that the user selects the "next" button 
+                worker.port.on('next_selected', function () { 
+                    // get the time elapsed since the last cta completion
+                    var timeElapsed = utils.getTimeElapsed(storageManager.get('lastSidebarCTACompleteTime'));
+
+                    // get the amount we need to add to make up for the difference between CTA completion and when the timer should be done
+                    var timeElapsedDiff = (intervals.defaultSidebarInterval - utils.getTimeElapsed(storageManager.get('lastSidebarCTACompleteTime')) > 0 ? intervals.defaultSidebarInterval - utils.getTimeElapsed(storageManager.get('lastSidebarCTACompleteTime')) : 0);
+
+                    // fast forward the lastSidebarCTACompleteTime to complete now
+                    storageManager.set('lastSidebarCTACompleteTime', timeElapsedDiff + timeElapsed);
+
+                    // set the sidebar props so we can advance sidebars
+                    module.exports.sidebarManager.setSidebarProps();
+                    // load the next sidebar with those sidebar props
+                    module.exports.sidebarManager.showSidebar(storageManager.get('sidebarProps'));
```

Comment:
I would turn the contents of this anonymous function into a separate function and simply call the function in each off the two `worker.port.on` blocks.